### DISCUSSION TITLE
Recognition of non-English characters

### DISCUSSION THREAD

**Discussion by psloplion** [open-mmlab/mmocr#1871] [Q&A]
Hello. I have created my own dataset for Ukrainian character recognition. Added symbols to the dictionary. For training, I used the CRNN model. But during training, the recall and precision always show 0.0000. When testing the model, predictions are not shown, but instead of them, puffy values. What could it be? Thanks.

**Comment by gaotongxiao** [comment]
Did you add Ukrainian characters to the dictionary as well?

And for the evaluation - now the evaluation metric only takes Chinese and English characters into account. Don't forget to add Ukranian character set to `valid_symbols` for the correct evaluation result: https://mmocr.readthedocs.io/en/dev-1.x/api/generated/mmocr.evaluation.metrics.WordMetric.html#wordmetric

**Comment by psloplion** [comment]
Do I understand correctly that you can add the valid_symbols line to the dict in the _base_ /default_runtime.py file here?
![Screenshot_select-area_20230421205356](https://user-images.githubusercontent.com/131407125/233703734-f1052a3e-61af-4c4e-a849-3ab7acce660d.png)

**Reply by psloplion** [reply]
Yes, I added all Ukrainian characters to the new file. Assembly files look like this:
![Screenshot_select-area_20230421210431](https://user-images.githubusercontent.com/131407125/233704763-ca394960-b0b5-4f8c-9abe-548fbd2c2fb1.png)

**Reply by gaotongxiao** [reply]
> Do I understand correctly that you can add the valid_symbols line to the dict in the _base_ /default_runtime.py file here? ![Screenshot_select-area_20230421205356](https://user-images.githubusercontent.com/131407125/233703734-f1052a3e-61af-4c4e-a849-3ab7acce660d.png)

Yes, and you should add the range of Ukrainian character's Unicode to `valid_symbol` in a way similar to Chinese characters (FYI, Chinese characters range from `\u4e00t` to `\u9fa5`):

```python
valid_symbol: str = '[^A-Z^a-z^0-9^\u4e00-\u9fa5]',
```

The implementation can be found here: https://mmocr.readthedocs.io/en/dev-1.x/_modules/mmocr/evaluation/metrics/recog_metric.html#WordMetric
